dashboard/bitcoin-grafana.json

```diff
@@ -940,17 +1094,39 @@
       }
     }
   ],
-  "refresh": "5m",
-  "schemaVersion": 20,
+  "refresh": "30s",
+  "schemaVersion": 16,
   "style": "dark",
   "tags": [
     "bitcoin"
   ],
   "templating": {
-    "list": []
+    "list": [
+      {
+        "allValue": null,
+        "current": {},
+        "datasource": "${DS_PROMETHEUS}",
+        "hide": 0,
+        "includeAll": false,
+        "label": null,
+        "multi": true,
+        "name": "node",
+        "options": [],
+        "query": "label_values(bitcoin_uptime, instance)",
```

Comment:
Because you're using the `instance` value here, this will only pull metrics from the most recent running [instance](https://prometheus.io/docs/concepts/jobs_instances/). This is probably fine if your target always has the same IP address and port, but it breaks down when run in other environments.

I have my bitcoin node running in kubernetes and therefore the `instance` value changes every time the exporter pod is restarted. Unfortunately that means I only see about an hours worth of data, vs the ~5 months of data that I know is stored in Prometheus.

I'm not sure what a better way to deal with that is. Probably additional labels on the scrape jobs.

Reply:
These are the labels I have on the `bitcoin_uptime` metric, for example. I would normally want to control this using the `app` label, but that's a kubernetes label and possibly not applicable to your setup.

```
bitcoin_uptime{app="bitcoin-exporter",instance="10.0.0.115:8334",job="kubernetes-pods",kubernetes_namespace="bitcoin",kubernetes_pod_name="bitcoin-exporter-85cd667696-ln5tz",pod_template_hash="85cd667696"}
```

Reply:
Ah, intersting...

I use hostnames in `~/.prometheus/prometheus.yml` and configure IP's in /etc/hosts (can be done dynamically too with dns) so the `instance` is a name that does not change followed by ":<port>"

```
cat ~/.prometheus/prometheus.yml
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

scrape_configs:
- job_name: 'node3'
  static_configs:
  - targets: ['b1:9100', 'l1:9100', 'b2:9100', 'b2:8334', 'l2:9100', 'base:9100', 'bl3:9100', 'bl4:9100', 'bl5:9100', 'bl5:8334', 'bl4:8334']
```

This would work if we document on how to configure prometheus. Yet, that assumes the port does not change.

Reply:
I played around with templating the variable (either `app` or `instance` when the dashboard is imported). Ultimately, I wasn't happy with how wide all graphs were with only one node.

https://github.com/jvstein/bitcoin-prometheus-exporter/tree/progress_with_label_templating